polaris-react/src/components/Box/Box.tsx

```diff
@@ -258,7 +246,13 @@ export const Box = forwardRef<HTMLElement, BoxProps>(
       '--pc-box-min-height': minHeight,
       '--pc-box-min-width': minWidth,
       '--pc-box-max-width': maxWidth,
-      '--pc-box-outline': outline ? `var(--p-border-${outline})` : undefined,
+      '--pc-box-outline-color': outlineColor
+        ? `var(--p-color-${outlineColor})`
+        : undefined,
```

Comment:
Same with above re: `--p-color-transparent` token. Probably need to update this too along the lines of below:
```suggestion
      // eslint-disable-next-line no-nested-ternary
      '--pc-box-outline-color': outlineColor
        ? outlineColor === 'transparent'
          ? 'transparent'
          : `var(--p-color-${outlineColor})`
        : undefined,
```

Reply:
Going to remove transparent from `outline` as I don't think it makes sense. A transparent outline could be added in the future if necessary.